File: src/proj.android/LongConnectLib/lconnectionlibrary/src/main/java/simple/bean/IMUser.java
package simple.bean;

import java.io.Serializable;

public class IMUser implements Serializable {
    public String uid;
    public int type = -1;//-1未知，0普通用户，1机器人
    public String avatar = "";//头像
    public String name = "";//昵称
    public int roleId = 0;//vip等级
    public int level = 0;//等级

    public IMUser(String uid, int type, String avatar, String name, int roleId, int level) {
        this.uid = uid;
        this.type = type;
        this.avatar = avatar == null ? "" : avatar;
        this.name = name == null ? "" : name;
        this.roleId = roleId;
        this.level = level;
    }

    public IMUser(String uid) {
        this.uid = uid;
    }

    @Override
    public String toString() {
        return "IMUser{" +
                "uid='" + uid + '\'' +
                ", type=" + type +
                ", avatar='" + avatar + '\'' +
                ", name='" + name + '\'' +
                ", roleId=" + roleId +
                ", level=" + level +
                '}';
    }
}

File: src/proj.ios/LCTest/LCTest/IMUser.m
//
//  IMUser.m
//  LCTest
//
//  Created by Happy on 2018/5/25.
//  Copyright © 2018年 lc. All rights reserved.
//

#import "IMUser.h"

@implementation IMUser

-(instancetype)init:(NSString*)uid withType:(int)type
{
    if(self=[super init]){
        self.uid = uid;
        self.type = type;
    }
    
    return self;
}

-(instancetype)init:(NSString*)uid withType:(int)type withAvatar:(NSString*)avatar 
	withName:(NSString*)name withRoleId:(int)roleId withLevel:(int)level
{
	if(self=[super init]){
        self.uid = uid;
        self.type = type;
        self.avatar = avatar;
        self.name = name;
        self.roleId = roleId;
        self.level = level;
    }
    
    return self;
}

-(NSString*)toString
{
    return [NSString stringWithFormat:@"IMUser{uid='%@', type=%d, avatar='%@', name='%@', roleId=%d, level=%d",self.uid,self.type,self.avatar,self.name,self.roleId,self.level];
}

@end
